### ISSUE TITLE
Query parameters not in router state when navigating with back button.

### ISSUE THREAD

**Issue by ngbrown** [FormidableLabs/redux-little-router#127]
I am running the most recent version, v12.1.1 and items that are parameters (`?q=test`) show up in `state.router.query` when navigating directly by url, or when pushing state.  However, when the back button is used after navigating to another route in the app, the `state.router.query` doesn't exist on `state.router`.

This is how I'm connecting to the parameters:

```js
const mapStateToProps = (state, ownProps) => {
  return {
    query: state.router.query.q
  }
};

export default connect(mapStateToProps)(Search)
```

Guarding for the existence of `state.router.query` doesn't help because it never shows up when using the back button.

This is what the Redux debugger shows when navigating forward to a link with a parameter:

![screenshot - 2016-12-05 2_12_08 am](https://cloud.githubusercontent.com/assets/228987/20879335/86751d0e-ba90-11e6-811e-84eac7020168.png)

This is what it shows when navigating back to the same page:

![screenshot - 2016-12-05 2_12_45 am](https://cloud.githubusercontent.com/assets/228987/20879344/91e97cca-ba90-11e6-804e-6ef568aa1865.png)

**Comment by ngbrown** [comment]
When using the back button, the action initiated on [store-enhancer.js#L67](https://github.com/FormidableLabs/redux-little-router/blob/master/src/store-enhancer.js#L67) doesn't include the `query` object while the action does contain `query` for other types of navigation.

At a minimum it looks like the following needs added:

```diff
+import qs from 'query-string';
...
    history.listen(newLocation => {
      /* istanbul ignore else */
      if (newLocation) {
        matchCache.clear();
+        newLocation.query = qs.parse(newLocation.search);
        store.dispatch(locationDidChange({
          location: newLocation,
          matchRoute
        }));
      }
    });
```

**Comment by tptee** [comment]
This is fixed in my WIP PR here: https://github.com/FormidableLabs/redux-little-router/pull/130

I'll ping when it's released!

**Comment by ngbrown** [comment]
Were you able to incorporate any of the other #128 items?  For example, I had also improved the tests and the coverage up to 100% by using `jsdom.changeURL()` instead of faking the `getLocation` override option.

**Comment by tptee** [comment]
Fixed a little early in `v12.1.2`! The solution in #130 is different (we avoid a double parse).
